### PR TITLE
feat(ci): Migrate dockerhub token to vault

### PR DESCRIPTION
**Pull request by chouetz**
### What does this PR do?
Fetch secret from vault instead of `asw ssm`

### Motivation
[acix-354](https://datadoghq.atlassian.net/browse/ACIX-354)

### Describe how to test/QA your changes
CI pipeline must be all green:
#### Modified jobs & status from [pipeline](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/pipelines/45492173)
docker_build_agent6
docker_build_agent6_arm64 -> This one is only generated on main, but shared definition with the agent7 version
docker_build_agent6_jmx
docker_build_agent6_jmx_arm64
docker_build_agent6_py2py3_jmx
image

docker_build_agent7
docker_build_agent7_arm64
docker_build_agent7_jmx
docker_build_agent7_jmx_arm64
image

docker_build_cluster_agent_amd64
docker_build_cluster_agent_arm64
docker_build_cws_instrumentation_amd64
docker_build_cws_instrumentation_arm64
docker_build_dogstatsd_amd64
docker_build_dogstatsd_arm64
docker_build_fakeintake
docker_build_ot_agent7
docker_build_ot_agent7_arm64
docker_build_ot_agent7_jmx
docker_build_ot_agent7_jmx_arm64
image

pull_test_dockers_arm64
pull_test_dockers_x64
image

####

### CODE REVIEW

**Review by brycekahle — APPROVED** [review]

**Review by spencergilbert — APPROVED** [review]

**Review by vboulineau — APPROVED** [review]

File: .gitlab/container_build/docker_linux.yml [hunk 1]
```diff
@@ -13,8 +13,8 @@
...
+    - $CI_PROJECT_DIR/tools/ci/fetch_secret.sh $DOCKER_REGISTRY_RO token | docker login --username "$DOCKER_LOGIN" --password-stdin "$DOCKER_REGISTRY_URL"
     - EXIT="${PIPESTATUS[0]}"; if [ $EXIT -ne 0 ]; then echo "Unable to locate credentials needs gitlab runner restart"; exit $EXIT; fi
     # Build image, use target none label to avoid replication
     - docker buildx build --no-cache --push --pull --platform linux/$ARCH --build-arg CIBUILD=true --build-arg GENERAL_ARTIFACTS_CACHE_BUCKET_URL=${GENERAL_ARTIFACTS_CACHE_BUCKET_URL} $BUILD_ARG --build-arg DD_GIT_REPOSITORY_URL=https://github.com/DataDog/datadog-agent --build-arg DD_GIT_COMMIT_SHA=${CI_COMMIT_SHA} --file $BUILD_CONTEXT/Dockerfile --tag ${TARGET_TAG} --label "org.opencontainers.image.created=$(date --rfc-3339=seconds)" --label "org.opencontainers.image.authors=Datadog <package@datadoghq.com>" --label "org.opencontainers.image.source=https://github.com/DataDog/datadog-agent" --label "org.opencontainers.image.version=$(inv agent.version)" --label "org.opencontainers.image.revision=${CI_COMMIT_SHA}" --label "org.opencontainers.image.vendor=Datadog, Inc." --label "target=none" $BUILD_CONTEXT
```
Comment (CelianR) [inline .gitlab/container_build/docker_linux.yml:20]: As there is `--push`, are you sure read only will work in this case?

File: .gitlab/container_build/fakeintake.yml [hunk 2]
```diff
@@ -15,8 +15,8 @@ docker_build_fakeintake:
     BUILD_CONTEXT: .
   script:
     # DockerHub login for build to limit rate limit when pulling base images
-    - DOCKER_LOGIN=$($CI_PROJECT_DIR/tools/ci/fetch_secret.sh $DOCKER_REGISTRY_LOGIN) || exit $?
-    - $CI_PROJECT_DIR/tools/ci/fetch_secret.sh $DOCKER_REGISTRY_PWD | docker login --username "$DOCKER_LOGIN" --password-stdin "$DOCKER_REGISTRY_URL"
+    - DOCKER_LOGIN=$($CI_PROJECT_DIR/tools/ci/fetch_secret.sh $DOCKER_REGISTRY_RO user) || exit $?
+    - $CI_PROJECT_DIR/tools/ci/fetch_secret.sh $DOCKER_REGISTRY_RO token | docker login --username "$DOCKER_LOGIN" --password-stdin "$DOCKER_REGISTRY_URL"
     - EXIT="${PIPESTATUS[0]}"; if [ $EXIT -ne 0 ]; then echo "Unable to locate credentials needs gitlab runner restart"; exit $EXIT; fi
     - docker buildx build --push --pull --platform ${PLATFORMS} --file ${DOCKERFILE} --tag ${TARGET} $BUILD_CONTEXT
```
Comment (CelianR) [inline .gitlab/container_build/fakeintake.yml:21]: Same comment here

File: .gitlab/container_build/docker_linux.yml (hunk 1 above)
Comment (chouetz) [reply .gitlab/container_build/docker_linux.yml:20]: Yes, all the jobs passed, as detailed on the description. IIUC all the "real" push actions are done on `public-images`

**Review by CelianR — APPROVED** [review]

**Pull request merged**